src/indexer.rs

```diff
@@ -221,43 +239,59 @@ impl Indexer {
             (first_block - 1, last_block, first_block - 1)
         };
 
+        // Get `current_height`
+        let current_height = if current_height == 0 {
+            if let Some(height) = self.forward_block {
+                height
+            } else {
+                0
+            }
+        } else {
+            current_height
+        };
+
         // Check, do we need fetch history data or force check from some block height
         let block = if self.force_index_from_block.is_some() || self.fetch_history {
-            if let Ok(block) = client.get_block(BlockKind::Height(num_height)).await {
+            if num_height > current_height {
+                println!(
+                    "Try to fetch block with height bigger than latest block. \
+                          Sleep: {FORWARD_BLOCK_TIMEOUT:?}"
+                );
+                sleep(FORWARD_BLOCK_TIMEOUT).await;
+                None
+            } else if let Ok(block) = client.get_block(BlockKind::Height(num_height)).await {
                 Some(block)
             } else {
                 // If block not found do not fail, just increment height
                 let mut data = self.data.lock().unwrap();
-                data.last_block = last_block + 1;
+                data.last_block = last_block;
                 None
             }
         } else {
             current_block
         };
-        let (height, chunks) = block?;
+        let (height, chunks, block_hash, prev_block_hash) = block?;
+
+        let last_block_hash = self.data.lock().unwrap().last_block_hash;
+        if let Some(block_hash) = last_block_hash {
+            if block_hash != prev_block_hash {
+                let mut data = self.data.lock().unwrap();
```

Comment:
Why do we need to update the last_block if it doesn't much? And what is the entire flow of this scenario?

Reply:
last_block is the index of the block we want to fetch now.
last_handled_block is the index of the block we have successfully fetched and processed.

We have now fetched the block with the index last_block, and it turned out that its predecessor is not last_handled_block. This means that we accidentally missed some block due to network errors. Therefore, we return the pointer to last_handled_block and try to fetch the blocks again.